doc/introduction-to-xkb.md

```diff
@@ -0,0 +1,200 @@
+      It may depend on:
+
+      - The _location_ and _language_ of the user, because languages may
+        require [specific keys][language input keys] for their input methods,
+        such as the _muhenkan_ key on Japanese keyboard and the _Hanja_ key
+        for Korean keyboards. The keyboard are usually classified by the
+        [standard][keyboard standard] it is based on, e.g. ANSI, ISO, JIS,
+        ABNT.
+      - The keyboard _vendor:_ keyboard may have a set of keys that are not
+        standard, or may be specific to an OS.
+      </dd>
+      <dt>Layout</dt>
+      <dd>
+      The identifier of the general layout to use. It usually refers to a
+      country or a language.
+      </dd>
+      <dt>Variant</dt>
+      <dd>
+      Any minor variants on the general layout. It may be national variants
+      </dd>
+      <dt>Options</dt>
+      <dd>
+      Set of extra options to customize the standard layouts.
+
+      Examples: switch modifiers keys, location of the compose key, etc.
+      </dd>
+    </dl>
+  </dd>
+  <dt>
+    KcCGST: <u>K</u>ey<u>c</u>odes, <u>C</u>ompat, <u>G</u>eometry,
+    <u>S</u>ymbols, <u>T</u>ypes @anchor KcCGST-intro
+  </dt>
+  <dd>
+    This is the low-level configuration of XKB and how the files are actually
+    organized in the _layout database_.
+    It is not really intuitive or straight-forward for the uninitiated.
+
+    @note _xkbcommon_ [does not propose an API for KcCGST](@ref KcCGST-support):
+    it is considered an implementation detail.
+    Instead, [RMLVO] is the preferred way for the user to configure XKB.
+
+    The KcCGST configuration consists of the following components:
+
+    <dl>
+      <dt>Key codes</dt>
+      <dd>
+      A translation of the raw [key codes] from the keyboard into
+      symbolic names.
+      </dd>
+      <dt>Compatibility</dt>
+      <dd>
+      A specification of what internal actions modifiers and various
+      special-purpose keys produce.
+      </dd>
+      <dt>Geometry</dt>
+      <dd>
+      A description of the physical layout of a keyboard.
+
+      @attention This legacy feature is [not supported](@ref geometry-support)
+      by _xkbcommon_.
+      </dd>
+      <dt>Key symbols</dt>
+      <dd>
+      A translation of symbolic key codes into actual [key symbols] (keysyms).
+      </dd>
+      <dt>Key types</dt>
+      <dd>
+      Types describe how a pressed key is affected by active [modifiers]
+      such as Shift, Control, Alt, etc.
+      </dd>
+    </dl>
+  </dd>
+  <dt>Complete Keymap @anchor keymap-intro</dt>
+  <dd>
+  A complete keymap is a _self-contained_ text file with all the [KcCGST]
+  components needed to define a layout. This is the result of the _resolution_
+  of the [RMLVO] and [KcCGST] configurations. This is also the format used by
+  X11 and Wayland when prompted to _serialize_ the layout in use.
+
+  @note This is a low-level configuration. [RMLVO] is the preferred way for the
+  end user to configure XKB.
```

Comment:
I would qualify this a bit. Some window managers do offer a way for a user to load a complete xkb file, and that's just fine for users who know XKB and want full control. That said, these days we also offer ways to plug custom components to the RMLVO database so maybe this is correct...

Reply:
Added a comment for power users